restapi/src/main/java/io/stargate/web/docsapi/service/write/DocumentWriteService.java

```diff
@@ -139,9 +140,48 @@ public DocumentWriteService(TimeSource timeSource, DocsApiConfiguration config)
       List<JsonShreddedRow> rows,
       boolean numericBooleans,
       ExecutionContext context) {
+    List<String> subDocumentPath = Collections.emptyList();
+    return updateDocument(
+        dataStore,
+        keyspace,
+        collection,
+        documentId,
+        subDocumentPath,
+        rows,
+        numericBooleans,
+        context);
+  }
+
+  /**
+   * Updates a single document, ensuring that existing document with the same key have the given
+   * sub-path deleted.
+   *
+   * @param dataStore {@link DataStore}
+   * @param keyspace Keyspace to store document in.
+   * @param collection Collection the document belongs to.
+   * @param documentId Document ID.
+   * @param subDocumentPath The sub-document path to delete.
+   * @param rows Rows of this document.
+   * @param numericBooleans If numeric boolean should be stored.
+   * @param context Execution content for profiling.
+   * @return Single containing the {@link ResultSet} of the batch execution.
+   */
+  public Single<ResultSet> updateDocument(
+      DataStore dataStore,
+      String keyspace,
+      String collection,
+      String documentId,
+      List<String> subDocumentPath,
+      List<JsonShreddedRow> rows,
+      boolean numericBooleans,
+      ExecutionContext context) {
+    // TODO for review
+    //  should we double check that each row begins with the given sub-path
+    //  ensuring correctness of the update
+
```

Comment:
@EricBorczuk what about this todo? passed `subDocumentPath` is only used here for the delete query.. It's expected that rows are already created with `subDocumentPath` in mind.. Basically it means you can call this method to update a document at the sub-path, but provide rows that point to some completely other place in the document..

WDYT?

Reply:
Yeah I think it's reasonable as a precondition to make sure that the sub path matches the rows' path

Reply:
Added. Will merge on green.